src/connection.rs

```diff
@@ -482,4 +569,60 @@ mod tests {
     fn arbitrary_recoverable_error() -> RequestError {
         RequestError::IO(std::io::Error::from(std::io::ErrorKind::UnexpectedEof))
     }
+
+    struct FakeBrokerConn {
+        conn: Box<dyn Fn() -> Result<Arc<FakeConn>> + Send + Sync>,
+    }
+
+    #[derive(Debug, PartialEq)]
+    struct FakeConn;
```

Comment:
The names confuse me a bit. What's the difference between a "fake broker conn(ection)" and a "fake conn(ection)"?

Reply:
Yeaaahhh I didn't put a whole lot of thought into the names 😅 I renamed `FakeBrokerConn` to `FakeBrokerRepresentation`, is that better?